Travis utilise install_zds.sh

A-312 opened the pull request:
Travis utilise le même script de install_zds (ref : https://github.com/zestedesavoir/zds-site/pull/5188#issuecomment-448971422 et https://github.com/zestedesavoir/zds-site/issues/5143#issuecomment-447053353)

Corrige : fix #5127 , fix: #5128 , fix #5265 , fix #5144 


Note : #5144 :  Gérer l'installation de jdk nous même est obligatoire pour avoir le même script d'installation avec travis.

# Prêt à Q/A

A-312 committed:
Travis utilise le même script que nous

A-312 committed:
Bypass la selection Humaine de l'OS...

A-312 committed:
Sans hack

A-312 committed:
il manque un 0

A-312 committed:
uncomment

A-312 commented:
# todo : 
 
 - [x] Avant le merge : Fold travis_script, export le script dans le header. :)
 - [x] Beautify before_install
 - [x] : Vérifier que les commandes make arrête le script en cas d'erreur : https://travis-ci.org/zestedesavoir/zds-site/builds/504397504

A-312 committed:
remet coveralls

A-312 committed:
Doublon avec ligne 191 "installing `virtualenv 16.2.0`..."

A-312 committed:
J'ai oublié de mettre --answer-yes

A-312 committed:
Affichage le message seulement si nécessaire

A-312 committed:
Ajout d'un saut à la ligne

A-312 committed:
Listen make error and exit 1, if we get one

A-312 commented:
Mon script fonctionne bien mais on a un problème extérieure avec les fixtures.

Le problème ne vient pas de ma PR car, il a été rencontré ici : https://zestedesavoir.com/forums/sujet/12174/installation-locale-du-zds/?page=1#p200207 avec l'ancien script d’installation

A-312 committed:
Change l'opérateur

A-312 committed:
Démarre zmd avant d'installer les fixtures

A-312 committed:
Positionne correctement la variable

A-312 commented:
J'ai activé zmd mais il génère trop de log

artragis commented:
> J'ai activé zmd mais il génère trop de log

pour l'instant tant pis, on corrigera plus tard.

A-312 commented:
> pour l'instant tant pis, on corrigera plus tard.

le principal problème c'est ça (trop long pour s'afficher sur travis) : 

![image](https://user-images.githubusercontent.com/18501150/54110233-2b9e2d80-43e1-11e9-81b2-31dbab4a2063.png)

artragis commented:
Ok, il faut donc augmenter le niveau de log du logger dans epub_utils

```diff
diff --git a/zds/tutorialv2/epub_utils.py b/zds/tutorialv2/epub_utils.py
index d28a62fe5..c1a53bc19 100644
--- a/zds/tutorialv2/epub_utils.py
+++ b/zds/tutorialv2/epub_utils.py
@@ -155,9 +155,10 @@ def build_ebook(published_content_entity, working_dir, final_file_path):
     build_content_opf(published_content_entity, chapters, images, ops_dir)
     build_container_xml(meta_inf_dir_path)
     build_nav_xhtml(ops_dir, published_content_entity, chapters)
-
+    zip_logger = logging.getLogger(__name__ + '.zip')
+    zip_logger.setLevel(logging.WARN)
     shutil.make_archive(str(final_file_path), format='zip', root_dir=str(Path(working_dir, 'ebook')),
-                        logger=logging.getLogger(__name__))
+                        logger=zip_logger)
     shutil.move(str(final_file_path) + '.zip', str(final_file_path))
```

A-312 committed:
Diminue le niveau de log

A-312 commented:
@artragis ça n'a rien changé : https://travis-ci.org/zestedesavoir/zds-site/jobs/504841723

artragis commented:
bah disons que les logs que tu avais cité ne sont plus là. donc si, ça a marché.
là ce sont d'autres logs. Et c'est au niveau de la config de l'appli qu'il faut réduire la verbosité pour mettre "INFO" plutôt que DEBUG

A-312 committed:
Test niveau log différent

A-312 committed:
revert

A-312 committed:
Diminue le niveau de log pour diminuer l'output

A-312 committed:
Nouveau système de log :)

A-312 committed:
Settings personnalisé

A-312 committed:
Settings personnalisé

A-312 committed:
Settings personnalisé

A-312 commented:
**artragis :** J'ai réduit le niveau des logs :) 

# Q/A : 

debian 9.8 : 

```
xzdec: success.

* Load virtualenv
* [+virtualenv] installing `virtualenv 16.2.0` with pip
./scripts/install_zds.sh: ligne 192: pip : commande introuvable
* [+virtualenv] creating virtualenv
Error: [Errno 30] Read-only file system: 'lib' -> '/home/a312/zds/zds-site/zdsenv/lib64'

* Load virtualenv
* activating venv `zdsenv`
./scripts/install_zds.sh: ligne 216: ./zdsenv/bin/activate: Aucun fichier ou dossier de ce type
!! No virtualenv, cannot continue
Makefile:7 : la recette pour la cible « install-linux-full » a échouée
make: *** [install-linux-full] Erreur 1
```

C'est pip3 et python3, sinon python retourne python2.7...

A-312 committed:
Change le titre de la catégorie

A-312 committed:
use pip3

A-312 commented:
https://github.com/pypa/virtualenv/issues/209#issuecomment-64207886

> upgrade virtualenv to the latest version(1.11.6), and use it as
> 
> ```
> virtualenv --always-copy testenv
> ```
> 
> the options will not use hard link but copy the files

Avoir zdsenv dans un sharedfolder avec Windows comme host n'est pas une bonne idée

A-312 committed:
unzip n'était pas installé pour tout le monde

A-312 committed:
Améliore la syntaxe

A-312 committed:
Améliore la gestion du virtualenv

A-312 committed:
Supprime le dossier sans confirmation fichier par fichier

A-312 committed:
Gulp n'est jamais installé en global par défaut

A-312 committed:
Fix "error: invalid command 'bdist_wheel'" car manquant dans setuptools

A-312 committed:
Suit la même norme

A-312 committed:
Anticipe une possible erreur lors de prochain changement
J'ai mis exVal à la fin de la ligne pour être sûr que personne ira
mettre du code entre la commande souhaitée et `$?`. Ce qui, si cela se
produit casserait la gestion des erreurs. :(

A-312 committed:
Support pour les VM (+ dossier partagé) avec Windows en host

A-312 committed:
Attend que realpath soit installé s'il ne l'est pas

A-312 committed:
Undo last commit (for shared folder)

A-312 committed:
j'ai push un changement qui devait rester en local :/

artragis commented:
https://travis-ci.org/zestedesavoir/zds-site/jobs/505458501#L2301 => "command coverage not found"

A-312 commented:
Surement le "retry_pip" qui bloque, il faudrait essayer "pip"

artragis commented:
non ça semble plutôt être le fait qu'avant on utilisait `pip install -r requirements-dev.txt`. Il semble manquer un `+prod` lorsque tu fais `./scripts/install_zds.sh --answer-yes --detect-os-version --travis-output $zds_install_argument`

D'ailleurs le nom `install_back_with_prod` est assez mal choisi c'est plutôt `install_mysql_dev_env`

A-312 commented:
> 
> 
> non ça semble plutôt être le fait qu'avant on utilisait `pip install -r requirements-dev.txt`. Il semble manquer un `+prod` lorsque tu fais `./scripts/install_zds.sh --answer-yes --detect-os-version --travis-output $zds_install_argument`
> 
> D'ailleurs le nom `install_back_with_prod` est assez mal choisi c'est plutôt `install_mysql_dev_env`

Normalement il est compris dans : $zds_install_argument

https://github.com/zestedesavoir/zds-site/pull/5299/files#diff-872477f61524f7941c1ab3437f47e96dR31

artragis commented:
> Normalement

pour vérifier, fais un `echo` avant alors

A-312 committed:
test avec echo

A-312 commented:
j'ai rajouté un echo

A-312 committed:
essaye avec pip sans travis_retry

artragis commented:
Ca change rien, mais j'ai investigué : c'est déjà comme ça aujourd'hui. Remets le travis_retry et je mergerai.

A-312 commented:
J'essaye deux choses avant, je pense savoir d'où vient le problème.

A-312 committed:
Test avec un autre environnement

A-312 committed:
Fix coverage

A-312 committed:
beautify l'affichage

A-312 committed:
fix erreur de syntax

A-312 committed:
unix2dos define_function.sh

A-312 committed:
Fix les permissions

A-312 committed:
Fix l'erreur de syntax

A-312 committed:
Fix line ending 2ème essai

A-312 committed:
add travis_output

A-312 committed:
Modifie la nomenclature des fichiers

A-312 committed:
Rajoute les includes là, où c'est nécessaire

A-312 committed:
export les variables

A-312 committed:
corrige une erreur de c/c dans echo (incidence visuelle)

A-312 commented:
Il faudra vérifier si `source ./scripts/ci_turbo.sh` est bien scopé quand c'est lié au front. Que l'export soit bien dans l'ensemble du script travis et pas seulement le fichier qui l'inclue.

A-312 committed:
J'ai oublié de définir la catégorie de zds_fold pour install_zds.sh

A-312 committed:
(test scope)

A-312 committed:
(undo test)

A-312 committed:
export ZDS_ENV

A-312 committed:
exit 1 si erreur

A-312 committed:
zds_fold_category n'utilisait pas la bonne variable

A-312 committed:
ZDS_ENV, 2ème essai

A-312 committed:
Déplace l'error handler dans le script parent

A-312 committed:
Apparement j'avais oublié de mettre le lint du frontend :')

Si l'avion casse la checklist...

A-312 committed:
Démarre zmd pour coverage

A-312 committed:
Ajout du chemin relatif, plutôt que la commande global

A-312 commented:
Le coverage se fait sur les 3 premiers tests, il faudrait peut etre le faire sur un seul. Il me semble inutile de le lancer 3 fois.

A-312 commented:
@artragis On tente ma PR comme elle est ?

En amélioration, il y aurait :
 
 - Réfléchir si l'installation de texlive via install_zds est intéressante, on pourrait utiliser l'ancien script de travis.yml pour gagner du temps ;
 - Gérer le temps de chaque tache "folder_travis" en sh (pour qu'il s'affiche dans les logs à droite de la ligne) ;
 - ~~Faire coverage une seule fois et non sur les 3 tests, l'enlever du premier test qui fait 18 min, diminuerait le temps nécessaire pour faire tourner travis.~~ ----> edit : impossible apparement chaque test fait des choses différentes avec coverage.

~~Il faudrait vérifier si ma modification (passage d'apt-get à pip) avec *python3-setuptools* et *libxml2-dev* n'est pas problématique, sous debian j'ai eu un problème avec le package "wheel" je ne sais pas si c'est lié. :/ J'hésite à le remettre, il faudrait tester à nouveau, (je reçois bientôt ma nouvelle carte graphique).~~ ---> edit : J'ai remis comme c'était avant, donc on installe les packages comme c'était avant cette pr.

artragis commented:
on va tenter comme ça, je te propose de faire un rebase pour être sûr que tout marche bien.

A-312 commented:
Ok je fais ça demain. J'aurais ma nouvelle carte graphique, je pourrais donc lancer la VM ! :)

A-312 committed:
Undo passage apt-get à pip. On remet comme c'était avant

A-312 commented:
C'est passé : https://travis-ci.org/zestedesavoir/zds-site/builds/522269334 / Je rebase

A-312 commented:
@artragis prêt à merge (si le test passe).

A-312 committed:
maj versions testées

artragis commented:
Le temps de build a vachement augmenté alors qu'on avait réussi à le faire baisser à 10 minutes environ. Je pense pas que du coup ça soit acceptable.

A-312 committed:
On installe pas tex

A-312 committed:
Accélère l'installation de latex

A-312 committed:
no fixture

A-312 commented:
Je pensais que : https://github.com/zestedesavoir/zds-site/pull/5299/commits/819184339b866853282ac90121b45b1128a48da2 allait être suffisant mais le temps d'installation reste aussi long. :( à voir ce que donne mon dernier commit https://github.com/zestedesavoir/zds-site/pull/5299/commits/28fcd348d61df1a270d4010bc69ffb1dfb40edd4

A-312 committed:
fix `texhash: command not found`

A-312 committed:
texhash est déjà exécuté directement après l'installation

A-312 commented:
@artragis Voila, rapide comme un avion !
